pkg/core/construct_graph.go

```diff
@@ -116,3 +121,82 @@ func (cg *ConstructGraph) FindUpstreamGateways(unit *ExecutionUnit) []*Gateway {
 	}
 	return gateways
 }
+
+func CreateConstructGraphFromFile(path string) (*ConstructGraph, error) {
+
+	type ConstructRepresentation struct {
+		Kind string `yaml:"kind"`
+	}
+
+	type EdgeRepresentation struct {
+		Source      string `yaml:"source"`
+		Destination string `yaml:"destination"`
+	}
+
+	type ConstructGraphRepresentation struct {
+		Nodes map[string]ConstructRepresentation `yaml:"nodes"`
+		Edges []EdgeRepresentation               `yaml:"edges"`
+	}
+
+	graph := NewConstructGraph()
+	input := ConstructGraphRepresentation{
+		Nodes: map[string]ConstructRepresentation{},
+	}
+	keys := map[string]AnnotationKey{}
+
+	f, err := os.Open(path)
+	if err != nil {
+		return graph, err
+	}
+	defer f.Close() // nolint:errcheck
+
+	err = yaml.NewDecoder(f).Decode(&input)
+	if err != nil {
+		return graph, err
+	}
+
+	for id, construct := range input.Nodes {
+		switch construct.Kind {
+		case "execution_unit":
+			unit := &ExecutionUnit{AnnotationKey: AnnotationKey{ID: id, Capability: annotation.ExecutionUnitCapability}}
+			graph.AddConstruct(unit)
+			keys[id] = unit.AnnotationKey
+		case "static_unit":
+			unit := &StaticUnit{AnnotationKey: AnnotationKey{ID: id, Capability: annotation.StaticUnitCapability}}
+			graph.AddConstruct(unit)
+			keys[id] = unit.AnnotationKey
+		case "orm":
+			unit := &Orm{AnnotationKey: AnnotationKey{ID: id, Capability: annotation.PersistCapability}}
+			graph.AddConstruct(unit)
+			keys[id] = unit.AnnotationKey
+		case "kv":
+			unit := &Kv{AnnotationKey: AnnotationKey{ID: id, Capability: annotation.PersistCapability}}
+			graph.AddConstruct(unit)
+			keys[id] = unit.AnnotationKey
+		case "redis_node":
```

Comment:
nit: missing `"redis_cluster'`?

Reply:
in ifc2 we dont support redis_cluster so i didnt add it